network/src/protocols/mod.rs

```diff
@@ -0,0 +1,456 @@
+pub(crate) mod discovery;
+pub(crate) mod feeler;
+pub(crate) mod identify;
+pub(crate) mod outbound_peer;
+pub(crate) mod ping;
+
+use crate::{
+    errors::{Error, PeerError},
+    peer_store::{Behaviour, Status},
+    peers_registry::RegisterResult,
+    NetworkState, PeerIndex, ProtocolContext, ProtocolContextMutRef, ServiceControl, SessionInfo,
+};
+use bytes::Bytes;
+use log::{debug, error, info};
+use p2p::{
+    builder::MetaBuilder,
+    service::{ProtocolHandle, ProtocolMeta},
+    traits::ServiceProtocol,
+    ProtocolId,
+};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+// TODO: change to u32 or u64
+pub type ProtocolVersion = u8;
+
+#[derive(Clone, Debug)]
+pub enum Severity<'a> {
```

Comment:
This struct is already deleted, should be a merge error?

Reply:
Not noticed, I'll remove it.